tests/test_binary.nim

```diff
@@ -839,6 +840,28 @@ proc testsForSync(binaryPath: static string) =
       let diff = gitDiffConcise(trackDir)
       check diff == expectedDiffOutput
 
+  # Don't leave cached prob-specs dir in detached HEAD state.
+  check git(["-C", psDir, "checkout", "main"]).exitCode == 0
+
+  if existsEnv("CI"):
+    suite "sync, without --offline":
+      test "can pull changes into cached prob-specs":
+        # Reset local `main` to a previous commit.
+        check git(["-C", psDir, "reset", "--hard",
+                  "0eda2318cb5622532e498559255c8fe141c9d07f"]).exitCode == 0
+
+        # Perform a sync without `--offline`.
+        execAndCheckExitCode(1, syncBase)
+
+        # Check that local HEAD and `main` point to same commit as `origin/main`.
+        let upstreamLatestRef = gitCheck(0, ["-C", psDir, "rev-parse",
+                                            "origin/main"])
```

Comment:
We don't need `getNameofRemote` here, because it's guaranteed to be named `origin` during CI.

Locally, a user might override `clone.defaultRemoteName` in their git config, so pulling from `origin/main` would fail.

I've thought about explicitly using `git clone --origin origin` in `cloneExercismRepo`, but that might surprise somebody.

Reply:
> I've thought about explicitly using git clone --origin origin in cloneExercismRepo, but that might surprise somebody.

Why would that surprise somebody?

Reply:
Because if somebody deliberately changes the `clone.defaultRemoteName` in their git config to be `foo`, they might reasonably assume that all newly cloned git repos on their machine have the remote called `foo`. It seems a little heavy-handed to override their explicit preference, even if we don't expect them to be running git commands in the cached repo.

And, thinking about this more, I think I don't want us to hard-code `git fetch origin/main` anyway. If somebody has no network connection, and hasn't run an online sync ever (or recently), they should be able to copy another local prob-specs to the cache location and have it stay working for a later online sync (even if the remote for `exercism/problem-specifications` is named e.g. `upstream`). Niche use case for a power user, sure, but it seems too fragile otherwise.

Reply:
Okay. I wouldn't mind it, but I'm fine with it